plugin/evm/header/block_gas_cost.go

```diff
@@ -59,3 +66,31 @@ func BlockGasCostWithStep(
 		timeElapsed,
 	)
 }
+
+// MinRequiredTip is the estimated minimum tip a transaction would have
+// needed to pay to be included in a given block (assuming it paid a tip
+// proportional to its gas usage). In reality, there is no minimum tip that
+// is enforced by the consensus engine and high tip paying transactions can
+// subsidize the inclusion of low tip paying transactions. The only
+// correctness check performed is that the sum of all tips is >= the
+// required block fee.
+//
+// This function will return nil for all return values prior to Subnet EVM.
+func MinRequiredTip(config *params.ChainConfig, header *types.Header) (*big.Int, error) {
+	if !config.IsSubnetEVM(header.Time) {
+		return nil, nil
+	}
+	if header.BaseFee == nil {
+		return nil, errBaseFeeNil
+	}
+	if header.BlockGasCost == nil {
+		return nil, errBlockGasCostNil
+	}
+
+	// minTip = requiredBlockFee/blockGasUsage
+	requiredBlockFee := new(big.Int).Mul(
+		header.BlockGasCost,
+		header.BaseFee,
+	)
+	return new(big.Int).Div(requiredBlockFee, new(big.Int).SetUint64(header.GasUsed)), nil
```

Comment:
nit it would be nicer split, this is a bit long to follow

```suggestion
	blockGasUsage := new(big.Int).SetUint64(header.GasUsed)
	return new(big.Int).Div(requiredBlockFee, blockGasUsage), nil
```

Reply:
This is same as in https://github.com/ava-labs/coreth/pull/808, I think we should keep it that way.